### ISSUE TITLE
Add new rule that makes users aware that ty disregards codes in `type: ignore` comments

### ISSUE THREAD

**Issue by noahklein00** [astral-sh/ty#2603]
### Summary

example.py:
```py
x: int = 5
x = None  # type: ignore[foo-bar]
```

```
$ ty check example.py
All checks passed!
```

adding any error code after a `type: ignore` doesn't cause failures in ty even if the error code is wrong or completely invalid.

However `ty: ignore` comments do trigger failures:
```py
x: int = 5
x = None  # ty: ignore[foo-bar]
```

```
error[invalid-assignment]: Object of type `None` is not assignable to `int`
 --> example.py:2:1
  |
1 | x: int = 5
2 | x = None # ty: ignore[foo-bar]
  | ~   ^^^^ Incompatible value of type `None`
  | |
  | Declared type `int`
  |
info: rule `invalid-assignment` is enabled by default

warning[ignore-comment-unknown-rule]: Unknown rule `foo-bar`
 --> example.py:2:23
  |
1 | x: int = 5
2 | x = None # ty: ignore[foo-bar]
  |                       ^^^^^^^
  |

Found 2 diagnostics
```

I'd expect the `type: ignore` comment to fail in the same way.

### Version

ty 0.0.13

**Comment by TurtleOrangina** [comment]
Could be intentional for compatibility with other tools that parse type: ignore, like mypy suppression? In the code bases I have switched to ty I have used `respect-type-ignore-comments = false` and converted all `type: ignore` comments to `ty: ignore`, this avoids the issue (and makes it clear for which exact tool these suppression messages are designed to work).

**Comment by AlexWaygood** [comment]
Indeed, this is intentional, so that it's possible to use ty in combination with other type checkers like mypy if you want to. ty doesn't look at the error code after a `type: ignore` comment at all (it treats all `type: ignore` comments as blanket suppression comments, unlike `ty: ignore` comments). Mypy, on the other hand, _only_ respects `type: ignore` comments; there's no `mypy: ignore` equivalent.

I wonder if we could expose a disabled-by-default setting where we would complain about error codes after `type: ignore` comments, though? Users who aren't interested in running multiple type checkers might find it helpful. Cc. @MichaReiser

**Comment by MichaReiser** [comment]
> I wonder if we could expose a disabled-by-default setting where we would complain about error codes after type: ignore comments, though? Users who aren't interested in running multiple type checkers might find it helpful.

I'd prefer a rule for this that makes users aware of the fact that ty ignores the rule codes entirely. I do think this rule would have to be on by default because this rule is really about users who don't know about this particular ty behavior and they also wouldn't know that they have to enable this rule :) Projects using another type checker can either set `respect-type-ignore-comments` to `false` or disable the rule to remove the warning.

**Comment by TurtleOrangina** [comment]
I would also like a setting for this, it would be a cleaner way to migrate a code base from mypy to ty, by setting something like `warn-on-unused-type-ignore-comments = true`. This way I would not risk forgetting to remove a type: ignore comment, and also developers that are used to ignore type errors with `type: ignore` don't have to learn a new way.

**Comment by MichaReiser** [comment]
ty already warns about unused `type: ignore` comments and this will be its own rule soon https://github.com/astral-sh/ruff/pull/22790

I don't recommend projects only using `ty` to use `type: ignore` unless they're okay with it being always blanked ignore. It never suppresses a specific ignore (and this is what the new rule that Alex suggested would teach users)

**Comment by TurtleOrangina** [comment]
I am still not happy with the state of `type: ignore` vs `ty: ignore`, and I think a more radical solution would be ideal, for which I have created a separate issue to discuss: #2767 

If we want a less radical solution, I would suggest at least to follow the suggestion of @AlexWaygood : `I wonder if we could expose a disabled-by-default setting where we would complain about error codes after type: ignore comments, though? Users who aren't interested in running multiple type checkers might find it helpful.`. But I would argue this rule should be *enabled* by default, since the more common scenario that ty should optimize for is that ty is used as the only type checker, and the rare situation of "multiple type checkers" should be supported but may require setting a config variable.

**Comment by carljm** [comment]
@TurtleOrangina I think you are proposing exactly the same thing as @MichaReiser -- an enabled-by-default rule which fires on any `type: ignore` with a rule code, to alert users that ty will not respect the rule code. Then users have the option to disable this rule (if using ty together with mypy), or switch to `ty: ignore`.

**Comment by TurtleOrangina** [comment]
@carljm That sounds good to me, I wasn't sure if the linked astral-sh/ruff#22790 was supposed to be this solution, since it is not. If we are indeed talking about a default enabled rule to warn about unused rule codes in comments like `# type: ignore[rule]`, then I am all for it!

**Comment by carljm** [comment]
I now feel that we should do #2928 instead of what is mentioned here.

**Comment by MichaReiser** [comment]
> I now feel that we should do [#2928](https://github.com/astral-sh/ty/issues/2928) instead of what is mentioned here.

I agree with this. I'll close this issue in favor of #2928
